src/Scripting/Core/Hosting/ObjectFormatter/CommonObjectFormatter.Builder.cs

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Scripting.Hosting
+{
+    /// <summary>
+    /// Object pretty printer.
+    /// </summary>
+    public abstract partial class CommonObjectFormatter
+    {
+        private sealed class Builder
+        {
+            private readonly StringBuilder _sb;
+
+            private readonly bool _insertEllipsis;
+
+            private readonly BuilderOptions _options;
+
+            private int _currentLimit;
+
+            public Builder(BuilderOptions options, bool insertEllipsis)
+            {
+                _sb = new StringBuilder();
+                _insertEllipsis = insertEllipsis;
+                _options = insertEllipsis ? options.SubtractEllipsisLength() : options;
+
+                _currentLimit = Math.Min(_options.MaximumLineLength, _options.MaximumOutputLength);
+            }
+
+            public bool LimitReached
+            {
+                get { return _sb.Length == _options.MaximumOutputLength; }
```

Comment:
Inherited code, but I had the same concern.  I was actually thinking of eliminating the property altogether.